src/main/java/org/testng/internal/IContainer.java

```diff
@@ -0,0 +1,22 @@
+package org.testng.internal;
+
+/**
+ * Represents the capabilities of a simple container to hold data
+ */
+public interface IContainer<M> {
+
+  /**
+   * @return - Retrieves data from the container
+   */
+  M[] getItems();
```

Comment:
Note for later: prefer to use a `Collection` because the modifications of the cache will be triggered in references.

Reply:
The cache has only 1 utilisation right now which is within TestRunner and in that context, the cache is never meant to be updated.